cmds/webboot/webboot.go

```diff
@@ -119,39 +123,83 @@ func (d *DirOption) exec(uiEvents <-chan ui.Event) (menu.Entry, error) {
 			entries = append(entries, iso)
 		}
 	}
+	entries = append(entries, &BackOption{})
 	return menu.DisplayMenu("Distros", "Choose an option", entries, uiEvents)
 }
 
-func main() {
-	flag.Parse()
-	if *v {
-		verbose = log.Printf
+// getCachedDirectory recognizes the usb stick that contains the cached directory from block devices,
+// and return the path of cache dir.
+// the cache dir should locate at the root of USB stick  and be named as "Images"
+// +-- USB root
+// |  +-- Images (<--- the cache directory)
+// |     +-- subdirectories or iso files
+// ...
+func getCachedDirectory() (string, error) {
+	blockDevs, err := block.GetBlockDevices()
+	if err != nil {
+		return "", fmt.Errorf("No available block devices to boot from")
 	}
 
-	entries := []menu.Entry{
-		&DownloadOption{},
+	mountPoints, err := ioutil.TempDir("", "temp-device-")
+	if err != nil {
+		return "", fmt.Errorf("Cannot create tmpdir: %v", err)
 	}
 
-	if *dir == "" {
-		mp, err := getCachedDirectory()
-		if err == nil {
-			entries = append(entries, &DirOption{
-				label: "Use Cached ISO",
-				path:  filepath.Join(mp.Path, "Image"),
-			})
+	for _, device := range blockDevs {
+		mp, err := mount.TryMount(filepath.Join("/dev/", device.Name), filepath.Join(mountPoints, device.Name), "", mount.ReadOnly)
+		if err != nil {
+			continue
+		}
+		cachePath := filepath.Join(mp.Path, "Images")
+		if _, err = os.Stat(cachePath); err == nil {
+			return cachePath, nil
 		}
-	} else {
-		entries = append(entries, &DirOption{
-			label: "Use Cached ISO",
-			path:  *dir,
-		})
 	}
+	return "", fmt.Errorf("Do not find the cache directory: Expected a /Images under at the root of a block device(USB)")
+}
 
-	entry, err := menu.DisplayMenu("Webboot", "Choose an ISO:", entries, ui.PollEvents())
+func getMainMenu(cacheDir string) menu.Entry {
+	entries := []menu.Entry{}
+	if cacheDir != "" {
+		// UseCacheOption is a special DirOption represents the root of cache dir
+		entries = append(entries, &DirOption{label: "Use Cached ISO", path: cacheDir})
+	}
+	entries = append(entries, &DownloadOption{})
+
+	entry, err := menu.DisplayMenu("Webboot", "Choose an option:", entries, ui.PollEvents())
 	if err != nil {
 		log.Fatal(err)
 	}
+	return entry
+}
+
+func goBackDir(currentPath string) menu.Entry {
+	backTo, _ := filepath.Split(currentPath)
```

Comment:
Does filepath.Dir not work for this?

Reply:
You are right. filepath.Split is exactly equals to filepath.Dir +filepath.Base, and i just want the Dir